File: app/src/main/java/com/jorbital/gymstat/viewmodels/RoutinesViewModel.java
package com.jorbital.gymstat.viewmodels;

import com.jorbital.gymstat.data.Routine;

import io.realm.OrderedRealmCollection;
import io.realm.Realm;
import io.realm.RealmResults;

public class RoutinesViewModel
{
    private Realm realm;
    private OrderedRealmCollection<Routine> allRoutines;

    public OrderedRealmCollection<Routine> getAllRoutines()
    {
        return allRoutines;
    }
    private void setAllRoutines(OrderedRealmCollection<Routine> allRoutines)
    {
        this.allRoutines = allRoutines;
    }

    public RoutinesViewModel()
    {

    }

    public void makeListOfRoutines()
    {
        RealmResults<Routine> routines = realm.where(Routine.class).findAll();

        setAllRoutines(routines);
    }
}

File: app/src/main/java/com/jorbital/gymstat/viewmodels/RoutinesViewModel.kt
package com.jorbital.gymstat.viewmodels

import android.arch.lifecycle.ViewModel

import com.jorbital.gymstat.data.Routine
import com.jorbital.gymstat.utils.LiveRealmData

import io.realm.Realm

import com.jorbital.gymstat.utils.routineModel

class RoutinesViewModel : ViewModel() {
    private val realm: Realm = Realm.getDefaultInstance()
    var allRoutines: LiveRealmData<Routine>? = null
        private set

    init {
        subscribeToAllRoutines()
    }

    private fun subscribeToAllRoutines() {
        allRoutines = realm.routineModel().findAllRoutines()
    }

    fun noRoutines(): Boolean? {
        if (allRoutines == null)
            return null
        return if (allRoutines?.results != null)
            allRoutines?.results!!.isEmpty()
        else
            null
    }

    override fun onCleared() {
        realm.close()
        super.onCleared()
    }
}
